AirCasting/SearchAndFollow/CompleteScreen/Views/Header/StaticSessionHeader.swift

```diff
@@ -53,7 +55,10 @@ private extension StaticSessionHeader {
     }
 
     func adaptTimeAndDate() -> Text {
-        let formatter = DateFormatters.SessionCartView.utcDateIntervalFormatter
+        var formatter: DateIntervalFormatter {
+            if userSettings.twentyFourHour { return DateFormatters.SessionCardView.utcDateIntervalFormatter }
+            return DateFormatters.SessionCardView.utcDateInterval12hFormatter
+        }
```

Comment:
I was thinking about that and wasn't convinced, because of the fact that we would have 'kind of tricky' `if else` in our `Formatters`. Those should be simple and not have additional logic beneath which user is not aware of at first use -- _just a comment of my mind process_. I will try to apply your approach ASAP!

Reply:
> I was thinking about that and wasn't convinced, because of the fact that we would have 'kind of tricky' `if else` in our `Formatters`. Those should be simple and not have additional logic beneath which user is not aware of at first use -- _just a comment of my mind process_. I will try to apply your approach ASAP!

That is true if we treat `DateFormatters` as a kind of "static library" (so to say). If it bothers you to put logic in there you can introduce another layer that will be used (Injected) here which will contain logic. It would be a bit cleaner, I agree. Both approaches are good for this project tho, you can pick whatever you feel like. As long as the logic is moved away from views :)